File: java/iforevents/src/main/java/com/iforevents/PageEvent.java
package com.iforevents;

import java.util.Collections;
import java.util.Date;
import java.util.LinkedHashMap;
import java.util.Map;

/** A page (web) or screen (mobile) view. */
public final class PageEvent {
    public final String name;
    public final Map<String, Object> properties;
    public final String navigationType;
    public final String toRoute;
    public final String previousRoute;
    public final Date timestamp;

    public PageEvent(String name, Map<String, ?> properties) {
        this(name, properties, null, null, null);
    }

    public PageEvent(String name, Map<String, ?> properties, String navigationType, String toRoute, String previousRoute) {
        this.name = name == null || name.isEmpty() ? "page_view" : name;
        this.properties = Collections.unmodifiableMap(new LinkedHashMap<String, Object>(properties == null ? Collections.<String, Object>emptyMap() : properties));
        this.navigationType = navigationType;
        this.toRoute = toRoute;
        this.previousRoute = previousRoute;
        this.timestamp = new Date();
    }
}

File: php/iforevents/src/PageEvent.php
<?php

declare(strict_types=1);

namespace IForevents;

/** A page (web) or screen (mobile) view. */
final class PageEvent
{
    public readonly string $name;
    public readonly \DateTimeImmutable $timestamp;

    /** @param array<string, mixed> $properties */
    public function __construct(string $name = 'page_view', public readonly array $properties = [], public readonly ?string $navigationType = null, public readonly ?string $toRoute = null, public readonly ?string $previousRoute = null)
    {
        $this->name = $name === '' ? 'page_view' : $name;
        $this->timestamp = new \DateTimeImmutable('now', new \DateTimeZone('UTC'));
    }
}
